### ISSUE TITLE
Crash using Network Link Conditioner - Very Bad Network

### ISSUE THREAD

**Issue by devedup** [aws-amplify/amplify-ui-swift-liveness#34]
On testing using 'Very bad network' on the iOS network link conditioner, it will throw an exception after you tap:

Begin verification > begin check. 

As soon as the countdown ends, it will crash. On turning off the network link conditioner, it works fine. 

Thread 54: "*** -[AVAssetWriter startWriting] Cannot call method when status is 1"

[Fitafy  27-06-2023, 09-02.crash.zip](https://github.com/aws-amplify/amplify-ui-swift-liveness/files/11879303/Fitafy.27-06-2023.09-02.crash.zip)

**Comment by devedup** [comment]
I'm alleviating the crash in [VideoChunker start] with

```
       guard state != .writing else {
            return 
        }
```

However, the underlying issue is still there. Likely a timing issue on the network call returning and the LivenessStateMachine I guess, as case .recording(ovalDisplayed: false): is called twice in FaceLivenessDetectionViewModel, when in 'Very bad network'.

**Comment by atierian** [comment]
Thanks for opening this. We'll investigate and follow up here with any updates.

**Comment by Merlin1stHere** [comment]
I faced the same crash if liveness check session is obsolete or invalid.

**Comment by atierian** [comment]
Resolved in release [v1.0.2](https://github.com/aws-amplify/amplify-ui-swift-liveness/releases/tag/1.0.2).
